### PR TITLE
Add file utility functions

### PR DESCRIPTION
**Pull request by tpoterba**

### CODE REVIEW

**Comment by konradjk** [comment]
Is it just me, or is it still missing in CI?

**Comment by konradjk** [comment]
Nevermind, needed to hard refresh.

File: src/main/scala/is/hail/utils/Py4jUtils.scala [hunk 1]
```diff
@@ -69,16 +69,32 @@ trait Py4jUtils {
...
 
+  private def readableBytes(bytes: Long): String = {
+    if (bytes < 1024)
+      bytes.toString
+    else if (bytes < 1024 * 1024)
+      formatDigits(bytes, 1024) + "K"
+    else if (bytes < 1024 * 1024 * 1024)
+      formatDigits(bytes, 1024 * 1024) + "M"
+    else
+      formatDigits(bytes, 1024 * 1024 * 1024) + "G"
```
Comment (konradjk) [inline src/main/scala/is/hail/utils/Py4jUtils.scala]: One more level, maybe? May as well future proof it now

File: python/hail/tests/test_utils.py [hunk 2]
```diff
@@ -54,6 +54,40 @@ def test_hadoop_methods(self):
...
+        self.assertFalse(hl.hadoop_exists(resource('doesnt.exist')))
+
+    def test_hadoop_is_file(self):
+        self.assertTrue(hl.hadoop_is_file(resource('ls_test/f_50')))
+        self.assertFalse(hl.hadoop_is_file(resource('ls_test/subdir')))
+
+    def test_hadoop_stat(self):
+        path1 = resource('ls_test')
+        stat1 = hl.hadoop_stat(path1)
+        self.assertEqual(stat1['is_dir'], True)
```
Comment (patrick-schultz) [inline python/hail/tests/test_utils.py]: Even if we can't test the values of all the fields, we could assert that the key set is what we expect. Also in hadoop_ls.

File: src/main/scala/is/hail/utils/Py4jUtils.scala [hunk 3]
```diff
@@ -51,6 +53,55 @@ trait Py4jUtils {
...
+      formatDigits(bytes, kilo) + "K"
+    else if (bytes < giga)
+      formatDigits(bytes, mega) + "M"
+    else if (bytes < tera)
+      formatDigits(bytes, giga) + "G"
+    else
+      formatDigits(bytes, tera) + "T"
+  }
+
+  def formatDigits(n: Long, factor: Int): String = {
```
Comment (patrick-schultz) [inline src/main/scala/is/hail/utils/Py4jUtils.scala]: Any reason this isn't private?

File: python/hail/tests/test_utils.py (hunk 2 above)
Comment (tpoterba) [reply python/hail/tests/test_utils.py]: good call.

File: src/main/scala/is/hail/utils/Py4jUtils.scala (hunk 3 above)
Comment (tpoterba) [reply src/main/scala/is/hail/utils/Py4jUtils.scala]: nope, will fix.

**Pull request merged**